easy_timezones/middleware.py

```diff
@@ -147,7 +148,11 @@ def process_request(self, request):
                         tz = lookup_tz_v2(ip)
 
         if tz:
-            timezone.activate(tz)
+            try:
+                timezone.activate(tz)
+            except pytz.exceptions.UnknownTimeZoneError:
```

Comment:
Why don't you want to check if timezone exists before applying it?
https://pypi.org/project/pytz/
`if timezone_str in pytz.all_timezones:`

Reply:
@amatsukov-sugarcrm  I already tested this way. Apparently `pytz.all_timezones` has `Asia/Famagusta` but Django  doesn't,

Reply:
Also python uses `EAFP` convention. https://docs.quantifiedcode.com/python-anti-patterns/readability/asking_for_permission_instead_of_forgiveness_when_working_with_files.html

Reply:
So I was thinking to do the same action as the timezone does't exist at all if Django doesn't support it.